crates/nargo_cli/src/cli/init_cmd.rs

```diff
@@ -24,33 +33,53 @@ fn test_main() {
 }
 "#;
 
+const LIB_EXAMPLE: &str = r#"fn my_util(x : Field, y : Field) -> bool {
+    x != y
+}
+
+#[test]
+fn test_main() {
+    assert(my_util(1, 2));
+
+    // Uncomment to make test fail
+    // assert(my_util(1, 1));
+}
+"#;
+
 pub(crate) fn run<B: Backend>(
     // Backend is currently unused, but we might want to use it to inform the "new" template in the future
     _backend: &B,
-    _args: InitCommand,
+    args: InitCommand,
     config: NargoConfig,
 ) -> Result<(), CliError<B>> {
-    initialize_project(config.program_dir);
+    let package_type = if args.lib { PackageType::Library } else { PackageType::Binary };
+    initialize_project(config.program_dir, package_type);
     Ok(())
 }
 
 /// Initializes a new Noir project in `package_dir`.
-pub(crate) fn initialize_project(package_dir: PathBuf) {
+pub(crate) fn initialize_project(package_dir: PathBuf, package_type: PackageType) {
     // TODO: Should this reject if we have non-Unicode filepaths?
     let package_name = package_dir.file_name().expect("Expected a filename").to_string_lossy();
     let src_dir = package_dir.join(SRC_DIR);
     create_named_dir(&src_dir, "src");
 
+    // TODO: Need to make type configurable
     let toml_contents = format!(
         r#"[package]
 name = "{package_name}"
+type = "{package_type}"
 authors = [""]
 compiler_version = "{CARGO_PKG_VERSION}"
 
 [dependencies]"#
     );
```

Comment:
Could we avoid strings and construct Struct instead?

Reply:
While making these changes, I felt like we'll probably need to adopt a more full-feature template generator tool like https://crates.io/crates/ffizer so I think we should just delay that decision.